src/field/pokemon.ts

```diff
@@ -276,7 +276,18 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
   isAllowedInBattle(): boolean {
     const challengeAllowed = new Utils.BooleanHolder(true);
```

Comment:
if you're adding an `isAllowed()` check, aren't these 2 lines redundant/removable?